frontend/components/SampleMutationRequest.tsx

```diff
@@ -0,0 +1,72 @@
+// Sample Mutation to be used for reference (for POST, PUT, DELETE)
+
+import React, { useState } from 'react';
+import axios from 'axios';
+import { Card, Text, Button } from 'tamagui';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+
+// Make POST request with axios
+const createPost = async (postData) => {
+  try {
+    const response = await axios.post(
+      'http://localhost:5000/api/posts',
+      postData,
+    );
+    return response.data;
+  } catch (error) {
+    return error.message;
+  }
+};
+
+export default function SampleMutationRequest() {
+  const [mutationResult, setMutationResult] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  const mutation = useMutation({
+    // mutationKey: ['UniqueNameSpecificToMutation'], // Optional: Descriptive key to identify this specific mutation
+    mutationFn: createPost, // Function that defines how to fetch data for this mutation
+  });
+
+  // on submit, send HTTP request
+  const handleSubmit = async () => {
+    setIsLoading(true);
+    setError(null);
```

Comment:
I believe that React Query already has functions for checking fails, errors, and other. Refer to the documentation for more help.